### PR TITLE
DOC nitpicks on the documentation of `tabular_learner`

### PR DESCRIPTION
**Pull request by glemaitre**
Those are comments that I wanted to raise regarding the documentation. There is nothing radical here. I just wanted to expand a bit the user guide by providing some information regarding the parameter of the `TableVectorizer`

### CODE REVIEW

File: skrub/_tabular_learner.py [hunk 1]
```diff
@@ -104,7 +114,7 @@ def tabular_learner(estimator, n_jobs=None):
     ...         "fasting_glucose": [35, 140, 44, 137],
     ...     }
     ... )
-    >>> y = [False, True, False, True]
+    >>> y = [0, 1, 0, 1]
```
Comment (glemaitre) [inline skrub/_tabular_learner.py:117]: I would not encourage to not use integral classification values when this is possible

File: skrub/_tabular_learner.py [hunk 2]
```diff
@@ -129,34 +136,34 @@ def tabular_learner(estimator, n_jobs=None):
...
 
-    By applying only the first pipeline step we can see the transformed data
-    that is sent to the supervised estimator (see the :obj:`TableVectorizer`
-    documentation for details):
+    By applying only the first pipeline step we can see the transformed data that is
+    sent to the supervised estimator (see the :obj:`TableVectorizer` documentation for
+    details):
 
-    >>> model.named_steps['tablevectorizer'].transform(X)            # doctest: +SKIP
+    >>> model.named_steps['tablevectorizer'].transform(X)
```
Comment (glemaitre) [inline skrub/_tabular_learner.py:143]: I would avoid not testing the docstring. I think this is useful to find out that this is outdated.

File: skrub/_tabular_learner.py [hunk 3]
```diff
@@ -129,34 +136,34 @@ def tabular_learner(estimator, n_jobs=None):
...
     Pipeline(steps=[('tablevectorizer', TableVectorizer()),
                     ('simpleimputer', SimpleImputer(add_indicator=True)),
                     ('logisticregression', LogisticRegression())])
 
-    We see that for the :obj:`~sklearn.linear_model.LogisticRegression` we get
-    the default configuration of the ``TableVectorizer`` which is intended to
-    work well for a wide variety of downstream estimators. Moreover, as the
-    ``LogisticRegression`` cannot handle missing values, an imputation step is
-    added.
+    We see that for the :obj:`~sklearn.linear_model.LogisticRegression` we get the
```
Comment (glemaitre) [inline skrub/_tabular_learner.py:157]: I wondering if the discussion here and below should not go directly to the user guide.

My own habit is to use this section to quickly copy pasting code to get started. If I want to get inside I would naturally go read the information in the user guide.

I was wondering if we want a similar pattern or this is just me?

File: skrub/_tabular_learner.py [hunk 4]
```diff
@@ -79,20 +83,26 @@ def tabular_learner(estimator, n_jobs=None):
...
+    :obj:`~sklearn.ensemble.HistGradientBoostingRegressor`.
+
+    >>> tabular_learner('classifier')
     Pipeline(steps=[('tablevectorizer',
                      TableVectorizer(high_cardinality_transformer=MinHashEncoder(),
                                      low_cardinality_transformer=ToCategorical())),
                     ('histgradientboostingclassifier',
                      HistGradientBoostingClassifier(categorical_features='from_dtype'))])
 
+    When requesting a ``'classifier'``, the last of the pipeline is set to a
```
Comment (TheooJ) [inline skrub/_tabular_learner.py:105]:
```suggestion
    When requesting a ``'classifier'``, the last step of the pipeline is set to a
```

File: skrub/_tabular_learner.py [hunk 5]
```diff
@@ -79,20 +83,26 @@ def tabular_learner(estimator, n_jobs=None):
...
-    >>> tabular_learner('regressor')                             # doctest: +SKIP
+    >>> tabular_learner('regressor')
     Pipeline(steps=[('tablevectorizer',
                      TableVectorizer(high_cardinality_transformer=MinHashEncoder(),
                                      low_cardinality_transformer=ToCategorical())),
                     ('histgradientboostingregressor',
                      HistGradientBoostingRegressor(categorical_features='from_dtype'))])
 
-    >>> tabular_learner('classifier')                            # doctest: +SKIP
+    When requesting a ``'regressor'``, the last of the pipeline is set to a
```
Comment (TheooJ) [inline skrub/_tabular_learner.py:95]:
```suggestion
    When requesting a ``'regressor'``, the last step of the pipeline is set to a
```

File: doc/encoding.rst [hunk 6]
```diff
@@ -10,27 +10,59 @@ Encoding or vectorizing creates numerical features from the data,
...
+baseline.
+
+The function :func:`tabular_learner` goes the extra mile by creating a machine-learning
+model that works well on tabular data. This model combines a :class:`TableVectorizer`
+with a provided scikit-learn estimator. Depending whether or not the final estimator
+natively support missing values, a missing value imputer step is added before the
+final estimator. The parameters of the :class:`TableVectorizer` are chosen based on the
+type of the final estimator.
+
+.. list-table:: Parameter values choice of :class:`TableVectorizer` when using :func:`tabular_learner` function
```
Comment (glemaitre) [inline doc/encoding.rst:34]: This table should look like:

image

File: doc/encoding.rst [hunk 7]
```diff
@@ -11,26 +11,56 @@ converting dataframes, strings, dates... Different encoders are suited
...
+   :header-rows: 1
+
+   * -
+     - ``RandomForest`` models
```
Comment (TheooJ) [inline doc/encoding.rst:38]: +1 for this, really useful

**Review by TheooJ — APPROVED** [review]

File: skrub/_tabular_learner.py (hunk 2 above)
Comment (glemaitre) [reply skrub/_tabular_learner.py:145]: OK now I understand why we were skipping :)

**Comment by glemaitre** [comment]
Now I understand why we were skipping the test.

I will have to change a bit the narrative. I think in the user guide, we have space to have an (*) to explain that the native support was added at some point. If this is not the case then we fallback to the same strategy than the forest.

This could avoid some user question. I'll do that tomorrow.

File: doc/encoding.rst (hunk 6 above)
Comment (jeromedockes) [reply doc/encoding.rst]: that's a good idea, super useful! the non-hist gradient boosting would go in the same column as randomforest but just "RandomForest" is good it's better to keep the headings short

**Comment by jeromedockes** [comment]
> Now I understand why we were skipping the test.

yes the choice depend on the scikit-learn version because in 1.2 the HGBT did not have the "from_dtype" option, and randomforest did not have missing value support so the output of the doctests would be a bit different. an alternative to skipping the test altogether could be adding ellipsis in the output

File: skrub/_tabular_learner.py [hunk 8]
```diff
@@ -48,19 +48,25 @@ def tabular_learner(estimator, n_jobs=None):
       missing values itself.
     - the last step is the provided ``estimator``.
 
-    **Note:** ``tabular_learner`` is a recent addition and the heuristics used
-    to define an appropriate preprocessing based on the ``estimator`` may change
-    in future releases.
+    Read more in the :ref:`User Guide <table_vectorizer>`.
+
+    .. note::
```
Comment (jeromedockes) [inline skrub/_tabular_learner.py:53]: :) I had done this at first but I thought the way notes are rendered with our template gives them too much visual importance for this comment

**Review by jeromedockes — APPROVED** [review]
thanks a lot @glemaitre !

**Comment by jeromedockes** [comment]
>  I'll do that tomorrow.

do you want to do that in this PR or should I merge it?
